File: leetcode/solutions/1221.SplitAStringInBalancedStrings/balancedStringSplit_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import balancedStringSplit from "./balancedStringSplit.ts";

Deno.test("1221. Split a String in Balanced Strings", () => {
  type test = [string, number];
  const tests: test[] = [
    ["RLRRLLRLRL", 4],
		["RLLLLRRRLR", 3],
		["LLLLRRRR", 1],
		["RLRRRLLRLL", 2],
  ];

  for (const [arg, want] of tests) {
    assertEquals(balancedStringSplit(arg), want);
  }
});

File: leetcode/solutions/1221.SplitAStringInBalancedStrings/balancedStringSplit_test.go
package solution

import (
	"testing"
)

func Test_1221_SplitAStringInBalancedStrings(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  string
		out int
	}{
		{"RLRRLLRLRL", 4},
		{"RLLLLRRRLR", 3},
		{"LLLLRRRR", 1},
		{"RLRRRLLRLL", 2},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := balancedStringSplit(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
